dask_ml/compose/_column_transformer.py

```diff
@@ -185,7 +185,13 @@ def _hstack(self, Xs):
                 warnings.filterwarnings("ignore", "Concatenating", UserWarning)
                 return dd.concat(Xs, axis="columns")
         elif da.Array in types:
-            return da.hstack(Xs)
+            # To allow compatibility with dask core 1.0.0, this is the
+            # definition of the dask.array.hstack inlined:
+            if all(x.ndim == 1 for x in Xs):
+                # tup, axis, allow_unknown_chunksizes
+                return da.concatenate(Xs, 0, True)
```

Comment:
It seems like this branch isn't tested. Could you add a test that hits it?

Reply:
@TomAugspurger: I am don't think that branch cannot actually be executed.  In order for it to hit that branch, we need all 1-D arrays or Series (to get `x.ndim == 1`).  But `sklearn.compose.ColumnTransformer` calls `_validate_output(self, result)`, which seems like it checks for 2-D objects:

```python
    def _validate_output(self, result):
        """
        Ensure that the output of each transformer is 2D. Otherwise
        hstack can raise an error or produce incorrect results.
        """
        names = [name for name, _, _, _ in self._iter(fitted=True,
                                                      replace_strings=True)]
        for Xs, name in zip(result, names):
            if not getattr(Xs, 'ndim', 0) == 2:
                raise ValueError(
                    "The output of the '{0}' transformer should be 2D (scipy "
                    "matrix, array, or pandas DataFrame).".format(name))  #  <-- LOOK HERE
```

Do you agree?

Reply:
If it can't be hit then let's remove it and the just have the else block.

Reply:
Will do.  I'll place a comment in there too.